File: warmup/service-lane/cpp/code.cpp

#include <cstdio>  
#include <cstring>  
#include <iostream>  
#include <algorithm>  
#include <map>  

using namespace std;

int main()  
{      
    
    int length, cases, i, j, min; 
    int* road = NULL;
    
    cin >> length >> cases;
    
    // Dynamically allocate an array to hold the road widths,
    // then load them into the array
    
    road = new int[length];
    
    for(int i = 0; i < length; i++){

	cin >> road[i];	
    }  
    
    while (cases--)  
    {  
	
	cin >> i >> j;  
	
	
	// Scan through the road segment looking for the minimum width. If
	// we hit width = 1, we can stop scanning because its not possible
	// to find a smaller value
	
	min = 3;

	for(i; i<=j; i++){
	    
	    if(road[i] < min){
		
		min = road[i];
	    }
	    
	    if(road[i] == 1){
		    break;
	    }   
	}
	
	cout << min << endl; 	
	
    }  
    
    return 0;  
} 

File: warmup/service-lane/python/code.py

# This problem is simple: scan through the road array and
# determine the minimum value
    
vars = map(int,raw_input().split())

length = vars[0]
cases = vars[1]

road = map(int,raw_input().split())

for i in range(0,cases):
    
    index = map(int,raw_input().split())
    i = index[0]
    j = index[1]
    
    print min(road[i:j+1])
